### PR TITLE
Feature/369420 run forms from db

### PR DESCRIPTION
**Pull request by davidjamesstone**

### CODE REVIEW

**Review by colinrotherham — COMMENTED** [review]
This looks ace!

Added some comments, but they're mainly me thinking out loud

Not blockers

File: src/api/forms/form-metadata-repository.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ export function list() {
 }
 
 /**
- * Retrieves a form metadata
+ * Retrieves a form metadata by id
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply src/api/forms/form-metadata-repository.js]: Fixed with b58b9bb399f3de26e9c05b396cc43bdbf33b80f0

File: src/api/forms/service.js [hunk 2]
```diff
@@ -72,7 +79,7 @@ export async function listForms() {
 }
 
 /**
- * Retrieves form metadata
+ * Retrieves form metadata by id
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply src/api/forms/service.js]: Fixed with b58b9bb399f3de26e9c05b396cc43bdbf33b80f0

File: src/api/forms/service.test.js [hunk 3]
```diff
@@ -50,7 +58,11 @@ describe('createForm', () => {
       title: 'Test form',
       organisation: 'Defra',
       teamName: 'Defra Forms',
-      teamEmail: 'defraforms@defra.gov.uk'
+      teamEmail: 'defraforms@defra.gov.uk',
+      draft: {
+        createdAt: date,
+        updatedAt: date
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply src/api/forms/service.test.js]: Fixed in 4cc0a3e73a73d919418b029bd521b9783abe78c4 to use `expect.any(Date)`; much cleaner, thanks.

File: src/api/forms/service.test.js [hunk 4]
```diff
@@ -59,6 +71,10 @@ describe('createForm', () => {
   })
 
   test('should create a new form without special characters in the name', async () => {
+    const date = new Date(2020, 1, 1)
+    jest.useFakeTimers()
+    jest.setSystemTime(date)
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply src/api/forms/service.test.js]: We could do yeah. For now I have remove fake timers in favour of `expect.any(Date)`

File: src/routes/forms.test.js [hunk 5]
```diff
@@ -55,7 +57,11 @@ describe('Forms route', () => {
     title: 'Test form',
     organisation: 'Defra',
     teamName: 'Defra Forms',
-    teamEmail: 'defraforms@defra.gov.uk'
+    teamEmail: 'defraforms@defra.gov.uk',
+    draft: {
+      createdAt: now,
+      updatedAt: now
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply src/routes/forms.test.js:63]: I'm looking forward to the new [Temporal API](https://tc39.es/proposal-temporal/docs/) landing in node/browsers

**Review by colinrotherham — APPROVED** [review]
Looks perfect, let's get it in 🚀

Ideally `git rebase origin/main` that merge commit away, but 🤷‍♂️

**Pull request merged**